adding .rst to docs/cdk (#114)
IMAP-Science-Operations-Center/sds-data-manager: docs/source/cdk/aws-setup.rst

****

> aws setup

Reviewer: Capitalize titles (AWS Setup)

> ****
> aws setup
> ****

Reviewer 2: ```suggestion
AWS setup
=========
```
You need to underline at least as long as the text itself.

> **Download/requirements:**

Reviewer 2: ```suggestion
Download/requirements
~~~~~~~~~~~~~~~~~~~~~~~
```
Make this a heading too, then it will show up in table of contents.

Ensure you have installed nodejs (newer than version 14), AWS CLI, and Docker

- `nodejs <https://nodejs.org/en/download/>`_

- `AWS CLI <https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html>`_

> - `Docker <https://docs.docker.com/get-docker/>`_

Reviewer 2: I don't think there should be a space between the entries.

> **New user**

Reviewer 2: Heading too

> - log-in to AWS console
> `AWS console <https://aws.amazon.com/console/>`_

Reviewer: ```suggestion
- log-in to `AWS console <https://aws.amazon.com/console/>`_
```

- use *imap-sdc-development* for Account ID
- enter user name and password

- select IAM from *Services* menu or search IAM
- select *Users* from left menu
- click your user button
- select *Security Credentials* tab
- click *Create Access Key*
- make note of *Access Key ID* and *Secret Access Key*
- you can download the .csv file if you want

> - click *Close*

Reviewer: Nitpick, but can you make this a numbered list?

You can have the numbers automatically generated by using `#.` instead of `-`

> 

Reviewer 2: Only need one blank line at the end.

AWS Setup

> =========
> Download/requirements
> ~~~~~~~~~~~~~~~~~~~~~~~

Reviewer 2: ```suggestion
=========

Download/requirements
~~~~~~~~~~~~~~~~~~~~~
```
Extra linebreak between the sections, and it is nice to make the underlines exactly the same length even though the spec does allow longer.

I would have said an extra linebreak after even the minor section underlines, but it doesn't look like @maxinelasp did that in imap-processing. I would follow her lead for consistency, which I think did include the linebreak between the primary page header.

Ensure you have installed nodejs (newer than version 14), AWS CLI, and Docker

- `nodejs <https://nodejs.org/en/download/>`_
- `AWS CLI <https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html>`_
- `Docker <https://docs.docker.com/get-docker/>`_

New User
~~~~~~~~~
#. Log-in to `AWS console <https://aws.amazon.com/console/>`_
#. Use *imap-sdc-development* for Account ID
#. Enter user name and password
#. Select IAM from *Services* menu or search IAM
#. Select *Users* from left menu
#. Click your user button
#. Select *Security Credentials* tab
#. Click *Create Access Key*
#. Make note of *Access Key ID* and *Secret Access Key*
#. You can download the .csv file if you want
#. Click *Close*

Existing User
~~~~~~~~~~~~~
#. In command line, run the following command to set up the aws environment::

    aws configure

> #. Enter the *Access Key ID* and *Secret Access Key* from the previous step

Reviewer 2: ```suggestion
#. Enter your *Access Key ID* and *Secret Access Key* (If you don't have these, follow the :ref:`aws-new-user` section)
```
Adding a restructured text link to the previous section would be nice too. https://www.sphinx-doc.org/en/master/usage/restructuredtext/roles.html#ref-role

```python
# just above the previous section
.. _aws-new-user:
```

Author: I cant seem to get this. It might be a "my.pars" or something to do with the conf.py file

> AWS Setup
> =========
> Download/requirements
> ~~~~~~~~~~~~~~~~~~~~~

Reviewer 2: ```suggestion
AWS Setup
=========

Download/requirements
~~~~~~~~~~~~~~~~~~~~~
```
Missed the newline here, but added it below. Here is an example of how Numpy's layout is with plenty of newlines between different entities and levels.
https://github.com/numpy/numpy/blob/48a8277855849be094a5979c48d9f5f1778ee4de/doc/source/index.rst?plain=1#L1-L7

Ensure you have installed nodejs (newer than version 14), AWS CLI, and Docker

- `nodejs <https://nodejs.org/en/download/>`_
- `AWS CLI <https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html>`_
- `Docker <https://docs.docker.com/get-docker/>`_

> .. _aws-new-user:
> New User

Reviewer 2: ```suggestion
.. _aws-new-user:

New User
```
I'd add a newline here as well just to immediately call out this reference link.

- `AWS CLI <https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html>`_
- `Docker <https://docs.docker.com/get-docker/>`_

New User
~~~~~~~~~
#. Log-in to `AWS console <https://aws.amazon.com/console/>`_
#. Use *imap-sdc-development* for Account ID
#. Enter user name and password
#. Select IAM from *Services* menu or search IAM
#. Select *Users* from left menu
#. Click your user button
#. Select *Security Credentials* tab
#. Click *Create Access Key*
#. Make note of *Access Key ID* and *Secret Access Key*
#. You can download the .csv file if you want
#. Click *Close*

Existing User
~~~~~~~~~~~~~
#. In command line, run the following command to set up the aws environment::

    aws configure

#. Enter the *Access Key ID* and *Secret Access Key* from the previous step
#. For region, set it to the AWS region you'd like to set up your SDS. For IMAP, we're using "us-west-2"::

    [imap]
    region=us-west-2
    aws_access_key_id=<Access Key>
    aws_secret_access_key=<Secret Key>

#. Then, you can set the profile used by cdk by setting the AWS_PROFILE environment variable to the profile name (in this case, imap)::

    export AWS_PROFILE=imap

> 

Reviewer: Nice! This whole file looks loads better
